lib/dal/src/materialized_view.rs

```diff
@@ -121,6 +121,78 @@ pub async fn build_all_mv_for_change_set(
     Ok(())
 }
 
+/// This function builds all Materialized Views (MVs) for the change set in the [`DalContext`].
+/// It assumes that there is an [`MvIndex`] for the change set.
+///
+/// If you are unsure what to use, look at [`build_all_mv_for_change_set`] and
+/// [`build_mv_for_changes_in_change_set`] instead.
+#[instrument(
+    name = "materialized_view.build_all_mv_for_change_set_and_existing_index",
+    level = "debug",
+    skip_all,
+    fields(
+        si.workspace.id = Empty,
+        si.change_set.id = %ctx.change_set_id(),
+    ),
+)]
+pub async fn build_all_mv_for_change_set_and_existing_index(
+    ctx: &DalContext,
+    frigg: &FriggStore,
+    kv_revision_for_existing_index: KvRevision,
+) -> Result<(), MaterializedViewError> {
+    let span = current_span_for_instrument_at!("debug");
+    span.record("si.workspace.id", ctx.workspace_pk()?.to_string());
+
+    // Pretend everything has changed, and build all MVs.
+    let changes = ctx
+        .workspace_snapshot()?
+        .map_all_nodes_to_change_objects()
+        .instrument(tracing::info_span!("Generating Changes for all nodes"))
+        .await?;
+
+    let (frontend_objects, patches) = build_mv_inner(
+        ctx,
+        frigg,
+        ctx.workspace_pk()?,
+        ctx.change_set_id(),
+        &changes,
+    )
+    .instrument(tracing::info_span!("Building MVs"))
+    .await?;
+
+    let index_entries = frontend_objects.into_iter().map(Into::into).collect();
+    let mv_index = MvIndex::new(ctx.change_set_id(), index_entries);
+    let mv_index_frontend_object = FrontendObject::try_from(mv_index)?;
+    frigg
+        .insert_object(ctx.workspace_pk()?, &mv_index_frontend_object)
+        .await?;
+
+    let patch_batch = PatchBatch {
+        meta: PatchBatchMeta {
+            workspace_id: ctx.workspace_pk()?,
+            change_set_id: Some(ctx.change_set_id()),
+            snapshot_from_address: None,
+            snapshot_to_address: Some(ctx.workspace_snapshot()?.id().await),
+        },
+        kind: PATCH_BATCH_KIND,
+        patches,
+    };
+
+    frigg
+        .update_index(
+            ctx.workspace_pk()?,
+            &mv_index_frontend_object,
+            kv_revision_for_existing_index,
+        )
+        .await?;
+
+    DataCache::publish_patch_batch(ctx, patch_batch)
+        .instrument(tracing::info_span!("Publishing patch batch"))
+        .await?;
+
+    Ok(())
+}
```

Comment:
This appears to be identical to `build_all_mv_for_change_set` except for `update_index` (and having the kv revision passed in)? Feels like we should try to minimize the duplication between them.